### DISCUSSION TITLE
Working well for me: 0.7.6-beta1

### DISCUSSION THREAD

**Discussion by cumiller** [Doraku/DefaultDocumentation#75] [General]
The 0.7.6-beta1 has been working well for me for the two issues I ran into: 
-Duplicate namespace entries in externLinks causing a key already added error
-Two spaces added at the end of each line messing up Mermaid diagrams

My two remaining issues are: 
`<br/>` tags in Mermaid diagrams are replaced with `<br />` (space inserted after "br". This fails to get interpreted in DevOps and displays as written: `<br />`.
   and
When parsing command line arguments, the parser fails to separate the multiple externLinks separated by `|`.

Have you had a chance to look into either of remaining two issues? I know the "br" solution I provided is sort of a hack and you were going to look for another option. For the externLinks paths, adding the "Separator" to the attribute seemed like a good fix.

**Comment by Doraku** [comment]
I closed most of the backlog that accumulated during my holyday break and created a new beta `0.7.6-beta2` so you can check if everything works for you.
For the `<br />` thing, after looking into it some more it appears it can't be resolved as easily as I thought, I would need to not use System.Xml.Linq and parse things manually so for now I just removed all extra spaces from self closing tag that are written in the documentation with an ugly Replace >_> I am just hoping there is not an other environment that NEED this space to parse the xml correctly! (honestly this should be reported to DevOps as a bug haha)
I just want to add some logs for debugging and better error reporting before publishing a proper 0.7.6 version, maybe sometimes this week.

**Comment by cumiller** [comment]
I'll give beta2 a try.

Yes. the `<br />`-extra space issue is unfortunate. Thank you for going ahead and adding a patch for that.

**Comment by cumiller** [comment]
Was there a fix put in for the double spaces at the ends of lines? I thought there was a fix for that but I am still seeing them in my mermaid diagrams.

**Reply by Doraku** [reply]
No I completely forgot this issue between the time we talked about it and when I got back >_< I will look back on our discussion

**Reply by Doraku** [reply]
Just created `0.7.6` version, you can now use `DefaultDocumentationIgnoreLineBreak` property to not transform line break into markdown line break. You can also use the `ignorelinebreak` attribute on most xml documentation element (`summary`, ...) to change this setting locally. Hopefully this allow you to produce your expected documentation :)

**Comment by cumiller** [comment]
I am testing with DefaultDocumentationIgnoreLineBreak set to True in the project file and am not seeing it taking affect - I still see the double spaces at then end of each line. I am trying to do some testing debugging the Console project. I have a breakpoint in MarkdownWriter.WriteText() and see _ignoreLineBreak false for some iterations. I am trying to figure out why it is true for some items and false for others.

**Comment by cumiller** [comment]
I think the problem is in XElementExtension.GetIgnoreLineBreak(). It looks like the success or failure of the TryParse() overwrites the value of ignoreLineBreak.

**Reply by Doraku** [reply]
ah yes you are probably right,  the condition is all wrong, it should be:
```csharp
public static bool? GetIgnoreLineBreak(this XElement element) => bool.TryParse(element.Attribute("ignorelinebreak")?.Value, out bool ignoreLineBreak) ? ignoreLineBreak : default(bool?);
```
I tested it with a too small case and didn't see the issue. Can you try with this fix and see if it works as expected?

**Reply by cumiller** [reply]
Yes, I'll try it.

**Reply by cumiller** [reply]
It appears to be working. 

In the debugger, when `ignoreLineBreak` is true, I am seeing that it remains `true` when the element attribute does not exist.

**Reply by Doraku** [reply]
version `0.7.7` is here with the fix, hopping it does it for you finally :)
